### PR TITLE
Add filters when client is initialized without New

### PR DESCRIPTION
**Pull request by anuptalwalkar**
Covered by existing tests.

### CODE REVIEW

File: modules/uhttp/client/client.go [hunk 1]
```diff
@@ -32,28 +32,35 @@ import (
...
+	}
 }
 
 // Do is a context-aware, filter-enabled extension of Do() in http.Client
 func (c *Client) Do(ctx fx.Context, req *http.Request) (resp *http.Response, err error) {
 	filters := c.filters
-	// TODO: Need a way to handle the case where Client is initialized without the NewClient method
-	// and some filters are set. Need to always include the tracing filter
-	if len(filters) == 0 {
+	if c.setDefaultFilters == nil || *c.setDefaultFilters == false {
```
Comment (madhuravi) [inline modules/uhttp/client/client.go]: nil check not necessary. it defaults to false.

File: modules/uhttp/client/client.go [hunk 2]
```diff
@@ -32,28 +32,35 @@ import (
...
+	_serviceName string
+	_defaultTrue = true
+)
 
 // Client wraps around a http client
 type Client struct {
 	*http.Client
-	filters []Filter
+	filters           []Filter
+	setDefaultFilters *bool
```
Comment (madhuravi) [inline modules/uhttp/client/client.go]: This name is a bit confusing. Maybe defaultFilters added? Also add a comment indicating what it does.

File: modules/uhttp/client/client.go [hunk 3]
```diff
@@ -32,28 +32,35 @@ import (
 	"golang.org/x/net/context/ctxhttp"
 )
 
-var _serviceName string
+var (
+	_serviceName string
+	_defaultTrue = true
```
Comment (madhuravi) [inline modules/uhttp/client/client.go]: nit: default seems to indicate that the default value is true. The default value is actually false. Not sure it needs a variable, you can just set inline.

File: modules/uhttp/client/client.go [hunk 4]
```diff
@@ -32,28 +32,34 @@ import (
...
+	}
 }
 
 // Do is a context-aware, filter-enabled extension of Do() in http.Client
 func (c *Client) Do(ctx fx.Context, req *http.Request) (resp *http.Response, err error) {
 	filters := c.filters
-	// TODO: Need a way to handle the case where Client is initialized without the NewClient method
-	// and some filters are set. Need to always include the tracing filter
-	if len(filters) == 0 {
+	if c.defaultFiltersAdded == false {
```
Comment (alsamylkin) [inline modules/uhttp/client/client.go]: Do we care that this is not thread safe?

Comment (anuptalwalkar) [inline modules/uhttp/client/client.go]: well, not really. Its not exported and only set/changed once.

**Pull request merged**